src/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { AggregationCursor, Document, FindCursor } from 'mongodb';
+import { pipeline as callbackPipeline, Readable, PassThrough } from 'stream';
+import { promisify } from 'util';
+
+import stream from './stream';
+import type { SchemaParseOptions, Schema, SchemaField } from './stream';
+import * as schemaStats from './stats';
+
+type MongoDBCursor = AggregationCursor | FindCursor;
+
+/**
+ * Convenience shortcut for parsing schemas. Accepts an array, stream or
+ * MongoDB cursor object to parse documents` from.
+ */
+async function parseSchema(
+  docs: Document[] | MongoDBCursor | Readable,
+  options?: SchemaParseOptions
+): Promise<Schema> {
+  // Shift parameters if no options are specified.
+  if (typeof options === 'undefined') {
+    options = {};
+  }
+
+  let src: Readable;
+  // MongoDB Cursors
+  if ('stream' in docs) {
+    src = docs.stream();
+    // Streams
+  } else if ('pipe' in docs) {
+    src = docs;
+    // Arrays
+  } else if (Array.isArray(docs)) {
+    src = Readable.from(docs);
```

Comment:
Since moving from `event-stream` package to `stream` package provided by node, note that if you are running this on a web application using webpack 4, the [polyfill in webpack 4](https://github.com/webpack/node-libs-browser/blob/master/package.json#L26) will not have this `Readable.from` method. 

The fix here is to move to webpack 5 and use  v3.0.0`stream-browserify` polyfill for the stream package.

Reply:
Thanks for the callout. It wasn't intended to make the package not usable in browser environments. We could probably use some tests for that. Creating a Compass ticket to link.